deepparse/download_tools.py

```diff
@@ -0,0 +1,312 @@
+import gzip
+import os
+import shutil
+import sys
+import warnings
+from pathlib import Path
+from typing import Dict, Annotated
+from urllib.request import urlopen
+
+import requests
+from bpemb import BPEmb
+from fasttext.FastText import _FastText
+from requests import HTTPError
+from urllib3.exceptions import MaxRetryError
+
+from .errors.server_error import ServerError
+
+BASE_URL = "https://graal.ift.ulaval.ca/public/deepparse/{}.{}"
+CACHE_PATH = os.path.join(os.path.expanduser("~"), ".cache", "deepparse")
+
+# Status code starting in the 4xx are client error status code.
+# That is Deepparse, server problem (e.g. Deepparse server is offline).
+HTTP_CLIENT_ERROR_STATUS_CODE = 400
+# Status code starting in the 5xx are the next range status code.
+NEXT_RANGE_STATUS_CODE = 500
+
+MODEL_MAPPING_CHOICES: Annotated[Dict[str, str], "model_type and model_filename"] = {
+    "fasttext": "fasttext",
+    "fasttext-attention": "fasttext_attention",
+    "fasttext-light": "fasttext",
+    "bpemb": "bpemb",
+    "bpemb-attention": "bpemb_attention",
+}
+
+
+def download_fasttext_magnitude_embeddings(cache_dir: str, verbose: bool = True, offline: bool = False) -> str:
+    """
+    Function to download the magnitude pretrained fastText model.
+
+    Return the full path to the fastText embeddings.
+    """
+
+    os.makedirs(cache_dir, exist_ok=True)
+
+    model = "fasttext"
+    extension = "magnitude"
+    file_name = os.path.join(cache_dir, f"{model}.{extension}")
+    if not os.path.isfile(file_name) and not offline:
+        if verbose:
+            print(
+                "The fastText pretrained word embeddings will be download in magnitude format (2.3 GO), "
+                "this process will take several minutes."
+            )
+        extension = extension + ".gz"
+        download_from_public_repository(file_name=model, saving_dir=cache_dir, file_extension=extension)
+        gz_file_name = file_name + ".gz"
+        with gzip.open(os.path.join(cache_dir, gz_file_name), "rb") as f:
+            with open(os.path.join(cache_dir, file_name), "wb") as f_out:
+                shutil.copyfileobj(f, f_out)
+        os.remove(os.path.join(cache_dir, gz_file_name))
+    return file_name
+
+
+def download_weights(model_filename: str, saving_dir: str, verbose: bool = True) -> None:
+    """
+    Function to download the pretrained weights of the models.
+    Args:
+        model_filename: The network type (i.e. fasttext or bpemb).
+        saving_dir: The path to the saving directory.
+        verbose (bool): Turn on/off the verbosity of the model. The default value is True.
```

Comment:
```suggestion
        verbose (bool): Either or not to be verbose during the download of a model. The default value is True.
```

Otherwise, the sentence felt like a negation: like turn off the verbose equal to True (true turn off the verbose) or False.